plugins/buildingplan/buildingplan_cycle.cpp

```diff
@@ -45,13 +46,38 @@ struct BadFlags {
     }
 };
 
+// cache of walkability groups that have citizens in them
+static std::unordered_set<uint16_t> accessible_walkability_groups;
+
+void update_walkability_groups(){
+    static int32_t update_timestamp = -1;
+
+    // ensure that we update at most once per tick
+    auto frame_counter = df::global::world->frame_counter;
+    if (frame_counter == update_timestamp)
```

Comment:
this will give incorrect results if a world is loaded, exited, and then a different world is loaded (since `frame_counter` is reset to 0 on world load). It would be safer if the `update_timestamp` were declared outside of this function, referenced as an `extern` in `buildingplan.h` (or `buildingplan.cpp`) and reset in `plugin_load_site_data` (where `cycle_timestamp` is reset)

Reply:
You mean when you load a world, never unpause (or at least not long enough to reach a cycle), and then load a different world, you have one tick where there is stale data? Nice catch, I certainly didn't test that.